tests/test_coordinates.py

```diff
@@ -650,50 +233,89 @@ def test___eq___differInPoints(
-    actual = coordinates.copy()
-    actual.get_sph()
-    actual.get_cart()
-    assert coordinates == actual
+def test___eq___differInWeights_notEqual():
+    coordinates = Coordinates(1, 2, 3, weights=.5)
+    actual = Coordinates(1, 2, 3, weights=0.0)
+    assert not coordinates == actual
+
+
+def test___eq___differInShComment_notEqual():
+    coordinates = Coordinates(1, 2, 3, comment="Madre mia!")
+    actual = Coordinates(1, 2, 3, comment="Oh my woooooosh!")
+    assert not coordinates == actual
 
 
-def test___eq___differInDomain_notEqual():
-    coordinates = Coordinates(1, 2, 3, domain='sph', convention='side')
-    actual = Coordinates(1, 2, 3, domain='sph', convention='front')
+def test___eq___differInShape():
+    coordinates = Coordinates([1, 1], 2, 3)
+    actual = Coordinates(1, 2, 3)
     assert not coordinates == actual
 
 
-def test___eq___differInConvention_notEqual():
-    coordinates = Coordinates(domain='sph', convention='top_elev')
-    actual = Coordinates(domain='sph', convention='front')
+def test___eq___differInShape_empty():
+    coordinates = Coordinates()
+    actual = Coordinates(comment="Oh my woooooosh!")
     assert not coordinates == actual
 
 
-def test___eq___differInUnit_notEqual():
-    coordinates = Coordinates(
-        [1, 1], [1, 1], [1, 1],
-        convention='top_colat', domain='sph', unit='rad')
-    actual = Coordinates(
-        [1, 1], [1, 1], [1, 1],
-        convention='top_colat', domain='sph', unit='deg')
-    is_equal = coordinates == actual
-    assert not is_equal
+def test_comment_setter_type_error():
+    """Test setting the comment with a non-string type."""
+    coords = Coordinates()
+    with pytest.raises(TypeError, match="comment has to be of type string"):
+        coords.comment = 123  # non-string type
 
 
-def test___eq___differInWeights_notEqual():
-    coordinates = Coordinates(1, 2, 3, weights=.5)
-    actual = Coordinates(1, 2, 3, weights=0.0)
-    assert not coordinates == actual
+def test_comment_setter_valid():
+    """Test setting the comment with a valid string."""
+    coords = Coordinates()
+    coords.comment = 'This is a valid comment'
+    assert coords.comment == 'This is a valid comment'
 
 
-def test___eq___differInShOrder_notEqual():
-    coordinates = Coordinates(1, 2, 3, sh_order=2)
-    actual = Coordinates(1, 2, 3, sh_order=8)
-    assert not coordinates == actual
+def test_comment_init():
+    """Test initializing Coordinates with a comment."""
+    coords = Coordinates(comment='Initial comment')
+    assert coords.comment == 'Initial comment'
 
 
-def test___eq___differInShComment_notEqual():
-    coordinates = Coordinates(1, 2, 3, comment="Madre mia!")
-    actual = Coordinates(1, 2, 3, comment="Oh my woooooosh!")
-    assert not coordinates == actual
+def test_comment_empty_string():
+    """Test setting the comment to an empty string."""
+    coords = Coordinates()
+    coords.comment = ''
+    assert coords.comment == ''
```

Comment:
Could be removed. Already tested in `test_comment_setter_valid()`

Reply:
not for empty strings